lgc/patch/NggPrimShader.cpp

```diff
@@ -854,6 +854,19 @@ void NggPrimShader::buildPrimShaderCbLayoutLookupTable() {
   }
 }
 
+// =====================================================================================================================
+// Calculate the dword offset of each item in the stream-out control buffer
+void NggPrimShader::calcStreamOutControlCbOffsets() {
+  assert(m_pipelineState->enableSwXfb());
+
+  m_streamOutControlCbOffsets = {};
+
+  for (unsigned i = 0; i < MaxTransformFeedbackBuffers; ++i)
+    m_streamOutControlCbOffsets.bufOffsets[i] = (offsetof(Util::Abi::StreamOutControlCb, bufOffsets[0]) +
+                                                 sizeof(Util::Abi::StreamOutControlCb::bufOffsets[0]) * i) /
+                                                4;
+}
```

Comment:
I believe you can just say `offsetof(..., bufOffsets[i])`.

Reply:
Previously, I got such error from Jenkens CI job although the build could succeed on my VS2022:
the value of ‘i’ is not usable in a constant expression
m_streamOutControlCbOffsets.bufOffsets[i] = offsetof(Util::Abi::StreamOutControlCb, bufOffsets[i]) / 4;

Reply:
I will retry your suggestions

Reply:
Still get the build fail on gcc: https://github.com/amdrexu/llpc/actions/runs/4798995726/jobs/8538040625

Reply:
Okay. I was going off this: https://godbolt.org/z/3bjcWT3qq - but maybe that's a Clang-specific extension.

Reply:
My Visual Studio can successfully compile this but gcc fails :( I will fall back to the verbose revision of this code piece.